### PR TITLE
Clarify constraints on parameters with default values

### LINKED ISSUE #1097: Default values for parameter resolution is not well specificied
Take:
```
extern void f(in bit a, in bit<3> b = 2, in bit<5> c);

void g()
{
  f(1, 1);
}
```
The current specifications does not mention the extern call f will not work when used with not naming the arguments in the call.

The following could be added to solve this issue:
```
When not using named arguments, only defaulted arguments at the end are takened into account.
```

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #1097

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -1661,6 +1661,24 @@ directions:
   See Section [#sec-invoke-actions] for further details.
 - Default parameter values are only allowed for 'in' or direction-less parameters;
   these values must evaluate to compile-time constants.
+- If a callable object (action, control, parser, extern method, etc.)
+  has a signature containing default values for parameters, and if the
+  parameters containing default values do *not*  all appear at the end of
+  the list of parameters, then the object can be invoked only by
+  using named arguments, to make the call easy to disambiguate.
```
Comment (apinski-cavium) [inline p4-16/spec/P4-16-spec.mdk:1668]: This wording is confusing I think.
Especially when the example below shows an example of where not using named arguments is still allowed.
I don't have better wording on this but we definitely should combine the example below even with the example under @optional section below this too. Or rather move the example of default arguments only to this section from that section too.

**Comment by mbudiu-vmw** [comment]
@apinski-cavium is this better?

**Comment by mbudiu-vmw** [comment]
Let me make another section

**Comment by apinski-cavium** [comment]
> @apinski-cavium is this better?

Yes that is much easier to understand. So +1 from me now.

**Review by rst0git — APPROVED** [review]
LGTM

**Review by jafingerhut — APPROVED** [review]
LGTM

**Pull request merged**